api/opencl_runtime_layer.txt

```diff
@@ -582,18 +551,28 @@ Otherwise, it returns one of the following errors in _errcode_ret_:
     CL_MEM_HOST_READ_ONLY and _flags_ specify CL_MEM_HOST_WRITE_ONLY, or if
     _buffer_ was created with CL_MEM_HOST_NO_ACCESS and _flags_ specify
     CL_MEM_HOST_READ_ONLY or CL_MEM_HOST_WRITE_ONLY.
-  * CL_INVALID_VALUE if value specified in _buffer_create_type_ is not
+  * CL_INVALID_VALUE if the value specified in _buffer_create_type_ is not
     valid.
   * CL_INVALID_VALUE if value(s) specified in _buffer_create_info_ (for a
     given _buffer_create_type_) is not valid or if _buffer_create_info_ is
     `NULL`.
-  * CL_INVALID_BUFFER_SIZE if _size_ is 0.
   * CL_MEM_OBJECT_ALLOCATION_FAILURE if there is a failure to allocate
     memory for sub-buffer object.
   * CL_OUT_OF_RESOURCES if there is a failure to allocate resources required
     by the OpenCL implementation on the device.
   * CL_OUT_OF_HOST_MEMORY if there is a failure to allocate resources
     required by the OpenCL implementation on the host.
+// These errors are specific to the CL_BUFFER_CREATE_TYPE_REGION parameter
+  * CL_INVALID_VALUE if the region specified by the *<<cl_buffer_region>>*
+    structure passed in _buffer_create_info_ is out of bounds in _buffer_.
+  * CL_INVALID_BUFFER_SIZE if the _size_ field of the *<<cl_buffer_region>>*
+    structure passed in _buffer_create_info_ is 0.
+// Jon - what is _context_ for this function? It's not a parameter. Is it the
+// context with which _buffer_ was created?
```

Comment:
Yes, exactly.

Reply:
I think this would be more clear if it said "no devices in the context associated with _buffer_", removing the italics which make it "context" look like a function parameter, and adding "the". It's a bit out of scope for this MR, but making that clarification to get the comment off the table.